lms/djangoapps/course_home_api/outline/views.py

```diff
@@ -443,3 +443,34 @@ def save_course_goal(request):
             })
         except Exception:
             raise UnableToSaveCourseGoal
+
+
+@api_view(['POST'])
+def unsubscribe_from_course_goal_by_token(request, token):
+    """
+    API calls to unsubscribe from course goal reminders.
+
+    Note that this does not require authentication - this view may be hit from an email on a different device than
+    normal or whatever. We should still be able to unsubscribe the user. Instead, we use a token in the email to
+    validate that they have permission to unsubscribe.
+
+    This endpoint is very tightly scoped (only unsubscribe: no subscribing, no PII) because it is unauthenticated.
+
+    **Example Requests**
+        POST api/course_home/v1/unsubscribe_from_course_goal/{token}
+
+    **Example Response Data**
+        {'course_title': 'Cats & Dogs In Canadian Media'}
+
+    Returns a 404 response if the token was not found. Otherwise, returns some basic course info. But no PII.
+    """
+    # First update the goal
+    goal = get_object_or_404(CourseGoal, unsubscribe_token=token)
+    goal.subscribed_to_reminders = False
+    goal.save()
+
+    # Now generate a response
+    course_overview = get_course_overview_or_404(goal.course_key)
```

Comment:
Should this check be before we update the goal? That way if either the goal or the course overview don't exist, we 404 right away? Or is that kind of already implicitly baked in that the goal would be the one to 404?

Reply:
I figured if we can successfully unsubscribe you, let's honor that and finish the unsubscribe? The course overview is just extra. (Also like, realistically it will never fail - course overviews are generated if not present - I think the error case is truly rare edge things like old-mongo or something.)

Reply:
Well it only generates the course overview if it finds it in the modulestore I believe. So not never. Truly invalid courses would still 404.

But sure. I don't have strong opinions. Just wanted to bring it up in case it was merely an oversight of writing vs an active decision.

Reply:
Yeah truly invalid courses would still 404, but hopefully we'd have already 404'd on the token by that point. But again, I'd rather unsubscribe if we can rather than refuse to unsubscribe the poor user if our database is borked for some reason.